libraries/libformula-ui/src/test/java/org/pentaho/openformula/ui/FormulaDocumentTest.java

```diff
@@ -57,4 +62,23 @@ public void testParseFunctionBad2() throws BadLocationException {
     assertEquals( "Length", str.length(), doc.getLength() );
     assertEquals( "Number of elements: ", 17, element.getElementCount() );
   }
+
+  public void testParseTextOpenParenthesisCorrectlyPRD5009()  {
+    FormulaDocument doc = new FormulaDocument();
+    final String str = "=IF((1=3);1;2)";
+    FormulaElement[] result = FormulaParser.parseText( doc, str );
+    assertEquals( "class of =", result[0].getClass().getName(), "org.pentaho.openformula.ui.model2.FormulaOperatorElement" );
```

Comment:
Instead of using text here, compare the class instances directly. Class instances are equal when loaded from the same classloader, something that is guaranteed to be true for JUnit runners. So kill "getName()" and use FormulaOperatorElement.class and so on ..

Reply:
Thanks Thomas, PR Updated. 
Tried not to import all those classes, but it definitely looks better this way.